File: examples/src/bin/delete_bucket.rs
use objectscale_client::client::ManagementClient;

fn main() {
    let endpoint = "https://10.225.108.189:443";
    let username = "root";
    let password = "Password123@";
    let insecure = true;

    let name = "test";
    let namespace = "osai0a9250592a131336";

    let objectstore_endpoint = "https://10.225.108.187:4443";

    let management_client =
        ManagementClient::new(endpoint, username, password, insecure).expect("management client");
    let mut objectstore_client = management_client
        .new_objectstore_client(&objectstore_endpoint)
        .expect("objectstore client");

    objectstore_client
        .delete_bucket(name, namespace, false)
        .expect("delete bucket");
    println!("Deleted bucket: {:?}", name);
}

File: python/examples/delete_bucket.py
#
# Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#

import objectscale_client

def main():
    endpoint = "https://10.225.108.217:4443"
    username = "root"
    password = "Password123!"
    insecure = True

    namespace = "ns1"
    bucket_name = "luis_bucket"

    try:
        client = objectscale_client.client.ManagementClient(endpoint, username, password, insecure)
        client.delete_bucket(bucket_name, namespace, False)
        print("Deleted bucket:", bucket_name)
    except Exception as e:
        print("Failed to delete bucket:", e)


if __name__ == '__main__':
    main()
